backend/src/apiserver/config/config.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/client"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/common"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/resource"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/server"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	"google.golang.org/grpc/codes"
+	"os"
+	"time"
+)
+
+// deprecated
+type deprecatedConfig struct {
+	Name        string
+	Description string
+	File        string
+}
+
+type configPipelines struct {
+	Name        string
+	Description string
+	File        string
+	// optional, Name is used for PipelineVersion if not provided
+	VersionName string
+	// optional, Description is used for PipelineVersion if not provided
+	VersionDescription string
+}
+
+type config struct {
+	// If pipeline version already exists and
+	// LoadSamplesOnRestart is enabled, then the pipeline
+	// version is uploaded again on server restart
+	// if it does not already exist
+	LoadSamplesOnRestart bool
```

Comment:
I'm curious if it's worth adding this boolean to the `configPipelines` struct instead. That way we can say only upload the "Iris" pipeline once and don't care if the user deletes it but other pipelines always load on startup.

Reply:
yeah I thought about this as well and thought it out of scope, the initial reason to keep it was to maintain backwards compatibility with the original approach which evidently didn't have this requirement, but open to adding it

Reply:
I guess it's fine globally for now.